src/core/layertree/qgslayertreegroup.h

```diff
@@ -55,6 +55,21 @@ class CORE_EXPORT QgsLayerTreeGroup : public QgsLayerTreeNode
      */
     void setName( const QString &n ) override;
 
+    /**
+     * Returns the group's id.
+     */
+    QString id() const;
+
+    /**
+     * Sets the group's id.
+     */
+    void setId( const QString &id );
```

Comment:
I don't think we should have this in the API. ID should be consistent in time.
I would rather add an optional argument to the constructor.

Reply:
:+1: 

I removed the setId method as you suggest.

Now the constructor looks like : 
```h
// in .h file
QgsLayerTreeGroup( const QString &name = QString(), bool checked = true, const QString id = QString() );
```

and
```cpp
// in .cpp file
QgsLayerTreeGroup::QgsLayerTreeGroup( const QString &name, bool checked, const QString id )
  : QgsLayerTreeNode( NodeGroup, checked )
  , mName( name )
  , mId( ( id.isNull() ) ? generateId( name ) : id )
{
  connect( this, &QgsLayerTreeNode::visibilityChanged, this, &QgsLayerTreeGroup::nodeVisibilityChanged );
}
```